### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by MrSayyod**
- Made the BooksForm component statefully.
- Implemented the handleChange method.
- Implemented handleSubmit method.
- Connected the Create_BOOK and REMOVE_BOOK action to the components.
- Added a new table column to the BooksList component.
- Implemented the handleRemoveBook method.

### CODE REVIEW

File: src/containers/BooksForm.js [hunk 1]
```diff
@@ -8,21 +13,48 @@ const BooksForm = () => {
     'Learning',
     'Sci-Fi',
   ];
+  const [state, setState] = useState({ title: '', category: '' });
```
Comment (rindrajosia) [inline src/containers/BooksForm.js]: - [x] [REQUIRED] The `category value` should be equal to `Action`

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -8,21 +13,48 @@ const BooksForm = () => {
...
+    event.preventDefault();
+    if (title === '' || category === '') {
+      error = true;
+    } else {
+      createBook({
+        id: Math.round(Math.random() * 100) + 1,
+        title,
+        category,
+      });
+      setState({ title: '', category: '' });
```
Comment (rindrajosia) [inline src/containers/BooksForm.js]: - [x] [REQUIRED] The `category value` should be equal to `Action`

File: src/containers/BooksForm.js [hunk 3]
```diff
@@ -8,21 +13,48 @@ const BooksForm = () => {
     'Learning',
     'Sci-Fi',
   ];
+  const [state, setState] = useState({ title: '', category: '' });
+  let error = false;
+
+  const handleChange = ({ target: { name, value } }) => {
+    setState({ ...state, [name]: value });
+  };
+  const { title, category } = state;
```
Comment (rindrajosia) [inline src/containers/BooksForm.js]: - [x] [REQUIRED] Please `move it inside handleSubmit `function under `event.preventDefault()`

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

- [x] There is a bug with your book form because I am obliged to choose a new category even if the category is already on the default option or on the last category that the user has chosen. Please check my inline comment in order to ix it.



_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/containers/BooksForm.js (hunk 1 above)
Comment (MrSayyod) [reply src/containers/BooksForm.js]: Done

File: src/containers/BooksForm.js (hunk 2 above)
Comment (MrSayyod) [reply src/containers/BooksForm.js]: Done

File: src/containers/BooksForm.js (hunk 3 above)
Comment (MrSayyod) [reply src/containers/BooksForm.js]: moved the line 22 `const { title, category } = state;` inside handleSubmit function

File: src/containers/BooksForm.js [hunk 4]
```diff
@@ -8,21 +13,48 @@ const BooksForm = () => {
...
+              onChange={handleChange}
+              value={state.title || ''}
+            />
           </label>
         </div>
         <div className="category-div">
           <label htmlFor="book-category">
             Category:
-            <select id="book-category">
+            <select id="book-category" onChange={handleChange} name="category">
```
Comment (rindrajosia) [inline src/containers/BooksForm.js]: - [x] [REQUIRED] To make sure that your APP works without bug, please change this line to: 
```JSx
<select id="book-category" value={state.category} onChange={handleChange} name="category">
```
The value of the select option will change after each update even if, the onChange event is not called.

**Review by rindrajosia — APPROVED** [review]
Hi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: src/containers/BooksForm.js (hunk 4 above)
Comment (MrSayyod) [reply src/containers/BooksForm.js]: Done

**Pull request merged**